lib/bank_account/saving.rb

```diff
@@ -0,0 +1,45 @@
+module BankAccount
+	
+
+	class Saving < Account
+
+		def initialize(id, initial_balance)
+			super	
+			if @initial_balance < 10 
+				raise ArgumentError, "an error has occurred. Balance must equal more than $10. Please make a deposit."
+			end	
+			@available_balance
+		end #initialize end
+
+
+		def withdraw(w_amount)  
+			trans_fee = 2
+			mandatory_balance = 10
+			if @current_balance < 12
+				raise ArgumentError, "I'm sorry, There are not enough funds to complete this transaction. Please try again and/or make a deposit. Thanks. \n $#{@current_balance} is currently avaialble."
+			else  @current_balance >= 12
+					@current_balance -= (w_amount + trans_fee)
+					if @current_balance < 10
+						@current_balance += (w_amount + trans_fee)
+						raise ArgumentError,"This withdrawl would have overdrafted your account.\n Please recalculate and make another request.\n Perhaps a deposit? Current Balance: $#{@current_balance}."
```

Comment:
Can we put this check up in the initial `if` block?
Is there a way to combine these concerns into a single "Can this withdrawal be made?" concern?